### ISSUE TITLE
[Question] Is it possible to get perks for a non instanced item?

### ISSUE THREAD

**Issue by jgayfer** [Bungie-net/api#229]
I'm attempting to display perks for an item that matches DestinyInventoryItemDefinition. I've looked the item up in the sqlite database, but the perks field is an empty list (this seems to be the case for most items). Is it possible to retrieve the persistent perks of a non instanced item?

**Comment by vpzed** [comment]
What kind of item is it?  I figured out how to get the perks for weapons.  In the case of weapons the perks are under "sockets".

Here is a "de-referenced" example snippet where my code added "app_" fields with the displayProperties.name value for each <whatever>Hash in an object that was important to understand the perks of the weapon.  You can see the "frame" and "sights" listed in this example, but the blocks continue for the other perks as well.

    'sockets': {'detail': 'Details',
              'socketCategories': [{'app_socketCategory': 'WEAPON PERKS',
                                    'socketCategoryHash': 4241085061,
                                    'socketIndexes': [0, 1, 2, 3]},
                                   {'app_socketCategory': 'WEAPON MODS',
                                    'socketCategoryHash': 2685412949,
                                    'socketIndexes': [4, 6]}],
              'socketEntries': [{'app_singleInitialItem': 'Precision Frame',
                                 'reusablePlugItems': [{'app_plugItem': 'Precision '
                                                                        'Frame',
                                                        'plugItemHash': 1636108362}],
                                 'singleInitialItemHash': 1636108362,
                                 'singleInitialRewardItemListHash': 0,
                                 'socketTypeHash': 3956125808},
                                {'app_singleInitialItem': 'Red Dot 2 MOA',
                                 'reusablePlugItems': [{'app_plugItem': 'Red '
                                                                        'Dot 2 '
                                                                        'MOA',
                                                        'plugItemHash': 2405638015},
                                                       {'app_plugItem': 'Cleanshot '
                                                                        'IS',
                                                        'plugItemHash': 13792279},
                                                       {'app_plugItem': 'Hitmark '
                                                                        'IS',
                                                        'plugItemHash': 13792278}],
                                 'singleInitialItemHash': 2405638015,
                                 'singleInitialRewardItemListHash': 0,
                                 'socketTypeHash': 1283453667},

**Comment by vpzed** [comment]
I forgot to add since the lookup table in the manifest isn't obvious for some of the hashes here is how to decode:

    sockets.socketEntries[list].socketTypeHash - sockettype table
    sockets.socketEntries[list].singleInitialItemHash - inventoryitem table
    sockets.socketEntries[list].reusablePlugItems[list].plugItemHash - inventoryitem table
    sockets.socketCategories[list].socketCategoryHash - socketcategory table

**Comment by jgayfer** [comment]
Weapons or armor. Interesting that it's under sockets... I will take another peek

**Comment by vpzed** [comment]
What I can't figure out is how to determine the "base stats" of a piece of armor.  Like the Mobility/Recovery/Resilience of the base piece of armor by looking at the manifest item data).  There are a few places were those stats are mentioned in the object but the values of the ones I have inspected have always been zero.  For weapons you can get the base stats you see in say DIM, but for armor I can't make the connection.

**Comment by jgayfer** [comment]
Ah, I found the perks! I had find which of the sockets were of the weapon or armor socket type via the socket categories, then use those indexes to determine which of the entries are the perks I want. Them I can use the `singleInitialItemHash` to look up the perk in `DestinyInventoryItemDefinition`. Much more work than I anticipated, but it works!

Thanks for pointing me in the right direction :)

I haven't been able to find mobility/recovery/resilience for single items either.
